features/auth/signup/Signup.test.tsx

```diff
@@ -351,4 +354,30 @@ describe("Signup", () => {
     render(<View />, { wrapper });
     await waitFor(() => expect(mockRouter.pathname).toBe(dashboardRoute));
   });
+
+  it("restores state from signup token", async () => {
+    mockRouter.setCurrentUrl(`${signupRoute}?token=test-token`);
+    getSignupFlowMock.mockResolvedValue({
+      flow_token: "test-token",
+      account_is_filled: true,
+    });
+
+    render(<View />, { wrapper });
+
+    expect(
+      await screen.findByText(t("auth:community_guidelines_form.header"))
+    ).toBeVisible();
+  });
+
+  it("handles bad signup token", async () => {
+    mockRouter.setCurrentUrl(`${signupRoute}?token=bad-token`);
+    getSignupFlowMock.mockRejectedValue({
+      error_messages: ["Invalid token"],
+      status_code: 400,
+    });
+
+    render(<View />, { wrapper });
+
+    await waitFor(() => expect(mockRouter.pathname).toBe(loginRoute));
```

Comment:
Probably more a product decision than a code one - but I wonder if it'd make more sense to redirect the user back to the home page instead, in case of a bad token? Or even not redirect the user so they can see the error and might figure they copy/pasted the URL wrong?

Reply:
I think the cases are
1. Expired token - In this case they can try to sign in again and get sent another email with a fresh token
2. Copy+pasted token from email incorrectly - currently no way to distinguish this from an expired token, so 1 applies here too
3. Existing user somehow ended up on the signup page without wanting to. In this case they should log in
4. New user somehow ended up in the signup page without wanting to. I think this is unlikely and not that harmful to send them to the login page (they can still click signup from there)

Reply:
I guess you're saying those who visits the sign up route with the token URL param is more likely to be an existing partial user as opposed to a completely a new user? I suppose that makes sense as I'd expect most to be using this route this way from clicking or copying the link in their email.

Fair point for case 4 too since the sign up button is in the nav bar across all unauthenticated pages 👍